network/address_test.go

```diff
@@ -27,6 +29,12 @@ func TestConnType(t *testing.T) {
 }
 
 func TestAddress(t *testing.T) {
+	localhostNetworkAddresses, _ := net.LookupHost("localhost")
+	sort.Strings(localhostNetworkAddresses)
+	facebookNetworkAddresses, _ := net.LookupHost("facebook.com")
+	sort.Strings(facebookNetworkAddresses)
+	googleNetworkAddresses, _ := net.LookupHost("google.com")
```

Comment:
Mhhhh... I'm not sure I like the idea of doing a DNS resolution for each test run:
 - It makes our whole test suite depending on the internet network connectivity
 - do we really want to fetch facebook and google addresses ?? I would have prefer some respectable institution like EPFL, EFF, and such... 

Usually, in these situations, I'd prefer to leverage a Golang interface:
```
type Resolver interface {
    LookupHost(string) ([]string,error)
}
```
One implementation would be just a proxy to `net.LookupHost` and another one would be a dummy implementation returning static (correct) values, previously fetched with a call to `net.LookupHost`.
The tests should use the latter.
What do you think ?